modules/contacts/server/controllers/contacts.server.controller.js

```diff
@@ -45,6 +48,30 @@ exports.add = function(req, res) {
         });
       }
 
+      // Check if contact already exists
+      Contact.findOne({
+        $or: [
+          { users: [ req.body.friendUserId, req.user._id ] },
+          { users: [ req.user._id, req.body.friendUserId ] }
+        ]
+      }).exec(function(err, existingContact) {
+        if (err) return done(err);
+
+        if (existingContact) {
+          // Contact already exists!
+          return res.status(409).json({
+            message: errorHandler.getErrorMessageByKey('conflict'),
+            confirmed: existingContact.confirmed
```

Comment:
> The response body SHOULD include enough information for the user to recognize the source of the conflict.
Recommended at http://www.restpatterns.org/HTTP_Status_Codes/409_-_Conflict

Hence, passing _confirmed_ status back in the body too. Could pass entire contact back, but this seems enough.